src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -1821,4 +1821,115 @@ private final void appendChars(CharSequence s, int off, int end) {
         }
         count += end - off;
     }
+
+    /**
+     * Appends {@code count} copies of the character {@code c} to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count}.
+     *
+     * @param c      character to append
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws StringIndexOutOfBoundsException  if the result overflows the buffer
+     */
+    public AbstractStringBuilder repeat(char c, int count) {
+        if (count < 0) {
+            throw new IllegalArgumentException("count is less than zero: " + count);
+        }
+        if (count == 0) {
+            return this;
+        }
+        ensureCapacityInternal(this.count + count);
+        if (isLatin1() && StringLatin1.canEncode(c)) {
+            int index = this.count;
+            while (count-- != 0) {
+                value[index++] = (byte) c;
+            }
+            this.count = index;
+        } else {
+            if (isLatin1()) {
+                inflate();
+            }
+            int index = this.count;
+            while (count-- != 0) {
+                StringUTF16.putCharSB(value, index++, c);
+            }
+            this.count = index;
+        }
+        return this;
+    }
+
+    private AbstractStringBuilder repeatNull(int count) {
+        if (count < 0) {
+            throw new IllegalArgumentException("count is less than zero: " + count);
+        } else if (count == 0) {
+            return this;
+        }
+        int offset = this.count;
+        appendNull();
+        int length = this.count - offset;
+        int valueLength = length << coder;
+        if ((Integer.MAX_VALUE - offset) / count < valueLength) {
+            throw new OutOfMemoryError("Required length exceeds implementation limit");
+        }
+        int limit = count * length;
+        ensureCapacityInternal(offset + limit);
+        String.repeatCopyRest(value, offset << coder, limit << coder, length << coder);
+        this.count = offset + limit;
+        return this;
+    }
+
+    /**
+     * Appends {@code count} copies of the specified {@code CharSequence} {@code cs}
+     * to this sequence.
+     * <p>
+     * The length of this sequence increases by {@code count} times the
+     * {@code CharSequence} length.
+     *
+     * @param cs     a {@code CharSequence}
+     * @param count  number of times to copy
+     *
+     * @return  a reference to this object.
+     *
+     * @since 21
+     * @throws IllegalArgumentException  if {@code count} is less than zero
+     * @throws StringIndexOutOfBoundsException  if the result overflows the buffer
+     */
+    public AbstractStringBuilder repeat(CharSequence cs, int count) {
+        if (cs == null) {
+            return repeatNull(count);
```

Comment:
Or just replace `cs = "null";` and fall through to the rest.

Reply:
Same.

Reply:
Same, don't add more code than necessary.